object.go

```diff
@@ -664,9 +692,15 @@ func (obs *obs) AddLink(name string, obj *ObjectInfo) (*ObjectInfo, error) {
 		Name: name,
 		Opts: &ObjectMetaOptions{Link: &ObjectLink{Bucket: obj.Bucket, Name: obj.Name}},
 	}
+	info := &ObjectInfo{Bucket: obs.name, NUID: nuid.Next(), ModTime: time.Now().UTC(), ObjectMeta: *meta}
 
 	// put the link object
-	return obs.Put(meta, nil)
+	err = publishMeta(info, obs.js)
```

Comment:
Similar to previous comment:
```go
if err := publishMeta(info, obs.js); err != nil {
    return err
}
```

This applies to the other call to `publishMeta()` in this file as well.